src/org/opendatakit/briefcase/export/ExportConfiguration.java

```diff
@@ -77,18 +83,28 @@ public static ExportConfiguration load(BriefcasePreferences prefs) {
         prefs.nullSafeGet(START_DATE).map(LocalDate::parse),
         prefs.nullSafeGet(END_DATE).map(LocalDate::parse),
         prefs.nullSafeGet(PULL_BEFORE).map(Boolean::valueOf),
-        prefs.nullSafeGet(PULL_BEFORE_OVERRIDE).map(PullBeforeOverrideOption::from)
+        prefs.nullSafeGet(PULL_BEFORE_OVERRIDE).map(PullBeforeOverrideOption::from),
+        prefs.nullSafeGet(FORM_NEEDS_PRIVATE_KEY).map(Boolean::valueOf)
     );
   }
 
-  public static ExportConfiguration load(BriefcasePreferences prefs, String keyPrefix) {
+  public static ExportConfiguration load(Optional<FormStatus> form, BriefcasePreferences prefs, String keyPrefix) {
+    //Determine if this form needs a private key
+    Boolean formNeedsPrivateKey = form.filter(formStatus -> ((BriefcaseFormDefinition) formStatus.getFormDefinition()).isFileEncryptedForm()
+            || ((BriefcaseFormDefinition) formStatus.getFormDefinition()).isFieldEncryptedForm()).map(value -> Boolean.TRUE).orElse(Boolean.FALSE);
+
+    //Note: If this pref value is Optional.empty() then this is the first time Briefcase is loading this form
+    Optional<Boolean> oldpref = prefs.nullSafeGet(keyPrefix + FORM_NEEDS_PRIVATE_KEY).map(Boolean::valueOf);
     return new ExportConfiguration(
         prefs.nullSafeGet(keyPrefix + EXPORT_DIR).map(Paths::get),
         prefs.nullSafeGet(keyPrefix + PEM_FILE).map(Paths::get),
         prefs.nullSafeGet(keyPrefix + START_DATE).map(LocalDate::parse),
         prefs.nullSafeGet(keyPrefix + END_DATE).map(LocalDate::parse),
         prefs.nullSafeGet(keyPrefix + PULL_BEFORE).map(Boolean::valueOf),
-        prefs.nullSafeGet(keyPrefix + PULL_BEFORE_OVERRIDE).map(PullBeforeOverrideOption::from)
+        prefs.nullSafeGet(keyPrefix + PULL_BEFORE_OVERRIDE).map(PullBeforeOverrideOption::from),
+        //Check if this forms FORM_NEEDS_PRIVATE_KEY pref is set. If not this is its first config
+        //Determine if it needs private key or not and set the pref depending on the form definition
+        oldpref.map(value -> Optional.of(value)).orElse(Optional.of(formNeedsPrivateKey))
```

Comment:
If you do the previous suggestions, this becomes `prefs.nullSafeGet(keyPrefix + FORM_NEEDS_PRIVATE_KEY).map(Boolean::valueOf)` (mirroring what's done in the other fields)